src/helpers.js

```diff
@@ -846,7 +846,7 @@ export function getHostDevSourceObject(dev) {
 
         source = { vendor, product, bus, device };
     } else {
-        throw new Error(`getHostDevSourceObject: unsupport device type '${dev.type}'`);
+        console.warn(`getHostDevSourceObject: unsupport device type '${dev.type}'`);
```

Comment:
This is also going to crash. `getHostDevElemBySource()` does not get along with `source === undefined `, you can't call `"x" in undefined`.